### PR TITLE
Ryan johnson api endpoints

### PR DESCRIPTION
**Pull request by rjohnson4444**
- Add api endpoint for external consumption

### CODE REVIEW

**Comment by toriejw** [comment]
We probably don't need some of the files like jobs_helper.rb, otherwise looks great! :+1:

**Comment by rjohnson4444** [comment]
Nice! Good catch, I'll get rid of those. Thanks!

File: Gemfile [hunk 1]
```diff
@@ -9,6 +9,10 @@ gem 'jquery-rails'
 gem 'jbuilder', '~> 2.0'
 gem 'sdoc', '~> 0.4.0', group: :doc
 gem 'feedjira'
+gem 'responders'
+gem 'active_model_serializers'
+gem 'will_paginate'
```
Comment (rrgayhart) [inline Gemfile]: I think we already have this gem in master now that @Kealii's code has been merged

File: app/controllers/api/v1/jobs_controller.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class Api::V1::JobsController < ApplicationController
+  # respond_to :json
```
Comment (rrgayhart) [inline app/controllers/api/v1/jobs_controller.rb]: We probably shouldn't leave commented code if we don't need it

File: app/serializers/company_serializer.rb [hunk 3]
```diff
@@ -0,0 +1,3 @@
+class CompanySerializer < ActiveModel::Serializer
+  attributes :id, :name
```
Comment (rrgayhart) [inline app/serializers/company_serializer.rb]: Nice usage of serializers!

File: config/routes.rb [hunk 4]
```diff
@@ -5,6 +5,11 @@
   # You can have the root of your site routed with "root"
   root 'home#index'
 
+  namespace :api do
+    namespace :v1 do
+      resources :jobs, only: [:index]
```
Comment (rrgayhart) [inline config/routes.rb]: Great scoping!

File: spec/controllers/api/v1/jobs_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+    it 'returns json default of 25 jobs' do
+      30.times { create(:job) }
+      get :index, format: :json
+
+      expect(response_body['jobs'].count).to eq(25)
+    end
+
+    it 'returns jobs with correct attributes' do
+      1.times { create(:job) }
```
Comment (rrgayhart) [inline spec/controllers/api/v1/jobs_controller_spec.rb]: Probably don't need the `1.times` part

File: spec/controllers/api/v1/jobs_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        expect(job['location']).to be_instance_of(String)
+        expect(job['posted_date']).to be_instance_of(String)
+        expect(job['remote']).to be false
+        expect(job['raw_technologies']).to be_instance_of(Array)
+        expect(job['company']).to be_instance_of(Hash)
+      end
+    end
+
+    it 'returns the id and name of the company the job belongs to' do
+      1.times { create(:job) }
```
Comment (rrgayhart) [inline spec/controllers/api/v1/jobs_controller_spec.rb]: See above

**Comment by rjohnson4444** [comment]
Yep on both accounts. I'll fix that asap

**Comment by rrgayhart** [comment]
Nice work! A few small comments for fixes.

I think you'll need to pull in master and fix some mergies!

File: spec/controllers/api/v1/jobs_controller_spec.rb (hunk 6 above)
Comment (rjohnson4444) [inline spec/controllers/api/v1/jobs_controller_spec.rb]: My thought was to test that the data we get back is the right type.

File: app/serializers/company_serializer.rb (hunk 3 above)
Comment (rjohnson4444) [inline app/serializers/company_serializer.rb]: :+1:

File: config/routes.rb (hunk 4 above)
Comment (rjohnson4444) [inline config/routes.rb]: :+1:

**Comment by rjohnson4444** [comment]
Yep! I'm in the process of pulling in master right now.

File: spec/controllers/api/v1/jobs_controller_spec.rb (hunk 6 above)
Comment (rrgayhart) [inline spec/controllers/api/v1/jobs_controller_spec.rb]: I think it's just that:
```
1.times { create(:job) }
```

is functionally the same as

```
create(:job)
```

with less codez!

Comment (rjohnson4444) [inline spec/controllers/api/v1/jobs_controller_spec.rb]: Yea I thought you were talking about the actual test, until I saw above! Totally should have deleted the message above.

**Comment by rrgayhart** [comment]
@rjohnson4444 - Looks great! How do you feel about squashing those last few commits down?

Ready to :ship: !

**Comment by rjohnson4444** [comment]
Yep I'm in the process of figuring that out right now. When I do, I'll merge! :+1:

**Comment by rrgayhart** [comment]
Kiiillin it!

**Pull request merged**